src/bundle/Kernel/BundleNameCompatibilityTrait.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Bundle\CompatibilityLayer\Kernel;
+
+use Ibexa\CompatibilityLayer\BundleResolver\BundleNameResolver;
+
+trait BundleNameCompatibilityTrait
+{
+    protected ?BundleNameResolver $bundleNameResolver = null;
+    protected string $compatibilityLayerBundleName = 'IbexaCompatibilityLayerBundle';
+
+    public function getBundle(string $name)
+    {
+        if (!isset($this->bundles[$this->compatibilityLayerBundleName])) {
+            return parent::getBundle($name);
+        }
+
+        if ($this->bundleNameResolver === null) {
+            $this->bundleNameResolver = new BundleNameResolver();
+        }
+
+        $newBundleName = $this->bundleNameResolver->resolve($name, true);
+        if ($newBundleName) {
```

Comment:
`resolve`'s return type is `?string`. Could we be more strict about that here?

```suggestion
        $newBundleName = $this->bundleNameResolver->resolve($name, true);
        if (null !== $newBundleName) {
```

or 

```suggestion
        $newBundleName = $this->bundleNameResolver->resolve($name, true);
        if (!empty($newBundleName)) {
```